### PR TITLE
Update kubernetes to newer version

### PR DESCRIPTION
**Pull request by happilymarrieddad**
apiVersion no longer beta
update to using ClusterIP for newer networking

### CODE REVIEW

**Comment by 0xsegfault** [comment]
@jondubois Hey John, 

I'll make those changes shortly. I only add the server and deployment to the name to make it easier for me to remember what's going on in our production environment. There is no need to do it here. Sorry about that. As far as ClusterIP is concerned, it's the way you're supposed to do internal networking in Kubernetes. The other kind is going to be deprecated like loadbalancer is. That's at least my understanding. Thanks!

**Comment by 0xsegfault** [comment]
oops... wrong account.. sorry lol

**Comment by happilymarrieddad** [comment]
@jondubois I fixed those changes you asked about. Thanks!

**Comment by happilymarrieddad** [comment]
@jondubois hey man I was thinking if the docker images shared the same persistent storage claim could you replicate state servers and remove the single point of failure? You could use ClusterIP to handle the communication. What do you think?

**Comment by jondubois** [comment]
scc-state doesn't use persistent storage; it's all in memory. scc-state is not really a single point of failure because SCC can keep working without it. There will only be partial disruption of the service if both scc-state and an scc-broker fail at the exact same time; and even if this unlikely scenario were to happen, the disruption should only last a few seconds.

I will update the scc-guide to remove the phrase 'single point of failure' because it sound like it's a problem but it's not. You always need to make a trade-off between consistency or availability and SCC leans slightly more towards consistency but it's still very resilient in terms of availability.

**Comment by jondubois** [comment]
@happilymarrieddad Your PR looks good now. It might take a bit of time to merge because I want to test it first.

**Comment by happilymarrieddad** [comment]
@jondubois no worries

File: kubernetes/scc-ingress.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ spec:
   # Note that raw (unencrypted) WebSockets may not work if you are behind a corporate proxy so
   # it is advised that you provide a cert for production and access the SCC service via https:// and wss://
   tls:
-  - secretName: scc-tls-credentials
+    - secretName: scc-tls-credentials
```
Comment (jondubois) [inline kubernetes/scc-ingress.yaml]: Is this correct?

**Comment by botzill** [comment]
Hi.

I would like to ask what is the state of this PR? I would like to test k8s part but would be great to have the updated part. 

Thx, great work.

**Comment by jondubois** [comment]
@happilymarrieddad Thanks for this contribution :)

Sorry it took so long; I was busy and didn't get the time to test it until recently.

After testing it on GKE, I changed it to expose the `socketcluster` (worker) service using `NodePort` otherwise it doesn't work on GKE. This could also be an issue on other platforms.

**Comment by happilymarrieddad** [comment]
@jondubois cool thanks! Yea, I don't know.. I use kops and AWS for production. I played around a little bit on GKE but it was too much magic for me. Thanks!

**Pull request merged**